src/plugins/analysis/ipc/internal/ipcAnalyzer/ipcAnalyzer.py

```diff
@@ -0,0 +1,380 @@
+                    referents = getReferents(ghidraAnalysis, function, callSiteAddress)
+                    relevantSources = [s for s in sources if s.getSeqnum().getTarget() in referents]
+
+                    argValues = []
+                    for index in range(1, len(args)+1):
+                        argument = analyzeFunctionCallSite(ghidraAnalysis, function, call, index, relevantSources)
+                        argValues.append(parseStaticData(argument))
+                    callArgs[call] = argValues
+
+                containsKeyValue = False
+                start = 0
+                for arg in argValues:
+                    start += 1
+                    if isinstance(arg, str) and keyString in arg:
+                        containsKeyValue = True
+                        indices = getFormatSpecifierIndices(keyString, arg)
+                        formatTypes = getFormatTypes(arg)
+                        break
+                if not containsKeyValue:
+                    continue
+                # filter wich arg indices are relevant
+                for i in indices:
+                    argument = argValues[start+i]
+                    if not isinstance(argument, list):
+                        argument = [argument]
+                    argument = flatten(argument)
+                    for arg in argument:
+                        if isinstance(arg, formatTypes[i]) and stringIsPrintable(arg):
+                            result.append(arg)
+        resultList.extend(result)
+    resultList = sorted(set(resultList))
+    return resultList
+
+def main():
+    """
+    Two use cases:
+    1. Resolves version format strings if a key_string file exists
+    2. Runs an IPC analysis on the provided binary
+    The result is saved in a json file
+
+    :return: int
+    """
+    args = getScriptArgs()
+    if len(args) != 1:
+        resultPath = os.getcwd()
+    else:
+        resultPath = args[0]
+    ghidraAnalysis = GhidraAnalysis()
+
+    # Resolve version format string
+    keyStringList = getKeyStrings(resultPath)
+    if keyStringList:
+        keyStringList = set(keyStringList)
+        resultList = resolveVersionFormatString(ghidraAnalysis, keyStringList)
+        writeToFile("ghidra_output.json", resultList, resultPath)
+        return 0
+ 
+    # IPC Analysis
+    print("\n###########################################################")
+    print("Analyzing: {}".format(currentProgram.getExecutablePath()))
+    sinkFunctionNames = ghidraAnalysis.sinkFunctionNames
+    sourceFunctionNames = ghidraAnalysis.sourceFunctionNames
+    ipc = {}
+    ipc["ipcCalls"] = {}
+    # Iterator over all functions in the program
+    functionManager = currentProgram.getFunctionManager()
+    functions = [func for func in functionManager.getFunctions(True)]
+    # Step 1. Check if the binary has at least one sink function
+    functionNames = [func.name for func in functions]
+    if (set(sinkFunctionNames) & set(functionNames)):
```

Comment:
I used `&` because I wanted the intersection of the [sets](https://docs.python.org/3/tutorial/datastructures.html#sets), which can also be achieved with `set1.intersection(set2)`. So originally this was what I wanted:
```python
if set(sinkFunctionNames).intersection(set(functionNames)) is not None:
```
But I've changed it to:
```python
sink_functions = [
        func for func in functions if func.name in ghidra_analysis.sink_function_names
    ]
    if len(sink_functions) >= 1:
```